homeassistant/components/thermostat/proliphix.py

```diff
@@ -0,0 +1,206 @@
+                oid, value = line.split('=')
+                const = OIDS.get(oid[3:])
+                if const:
+                    self._data[const] = value
+        logger.debug("PDP collected data %s" % self._data)
+
+    def _set(self, **kwargs):
+        data = {}
+        for key, value in kwargs.items():
+            oid = _get_oid(key)
+            if oid:
+                data["OID%s" % oid] = value
+        url = "http://%s/pdp" % self._host
+        form_data = urlencode(data)
+        form_data += "&submit=Submit"
+        requests.post(url, auth=(self._user, self._passwd), data=form_data)
+
+    @property
+    def cur_temp(self):
+        return float(self._data['AverageTemp']) / 10
+
+    @property
+    def setback_heat(self):
+        return float(self._data['SetbackHeat']) / 10
+
+    @setback_heat.setter
+    def setback_heat(self, val):
+        self._data['SetbackHeat'] = int(val * 10)
+        self._set(SetbackHeat=self._data['SetbackHeat'])
+
+    @property
+    def hvac_state(self):
+        return int(self._data['HvacState'])
+
+    @property
+    def name(self):
+        return self._data['DevName']
+
+    @property
+    def fan_state(self):
+        if self._data['FanState'] == "2":
+            return "On"
+        else:
+            return "Off"
+
+
+class ProliphixThermostat(ThermostatDevice):
+    """ Represents a Proliphix thermostat. """
+
+    def __init__(self, host, user, passwd):
+        self._pdp = PDP(host, user, passwd)
+        # initial data
+        self._pdp.update()
+        self._name = self._pdp.name
+
+    @property
+    def should_poll(self):
+        return True
+
+    def update(self):
+        self._pdp.update()
+
+    # change this so that we graph more often and collect the
+    # current_temperature when it changes.
+    @property
+    def state(self):
+        return self.current_temperature
+
+    @property
+    def name(self):
+        """ Returns the name. """
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        return {
+            "fan": self._pdp.fan_state
+        }
+
+    @property
+    def unit_of_measurement(self):
+        """ Returns the unit of measurement. """
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_temperature(self):
+        """ Returns the current temperature. """
+        return self._pdp.cur_temp
+
+    @property
+    def target_temperature(self):
+        """ Returns the temperature we try to reach. """
+        return self._pdp.setback_heat
+
+    @property
+    def operation(self):
+        state = self._pdp.hvac_state
+        if state in (1, 2):
+            return STATE_IDLE
+        elif state == 3:
+            return STATE_HEAT
+        elif state == 6:
+            return STATE_COOL
+        return STATE_IDLE
+
+    def set_temperature(self, temperature):
+        """ Set new target temperature. """
+        self._pdp.setback_heat = temperature
+
+    @property
+    def min_temp(self):
```

Comment:
I see what's going on . Instead of fixing it here, let's fix it in the thermostat class so it is fixed for all. Rounding off [these values](https://github.com/balloob/home-assistant/blob/dev/homeassistant/components/thermostat/__init__.py#L224-L232) should fix it.